Calculate/Calculator.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Numerics;
+
+
+namespace Calculate;
+
+public class Calculator <T> where T : INumber<T> {
+    
+    public IReadOnlyDictionary<char, Func<T, T, T>> MathematicalOperations { get; } =
+        new Dictionary<char, Func<T, T, T>>
+        {
+            ['+'] = CalculatorOperations.Add,
+            ['-'] = CalculatorOperations.Subtract,
+            ['*'] = CalculatorOperations.Multiple,
+            ['/'] = CalculatorOperations.Divide 
+        };
+        
+    public bool TryCalculate(string calculation, out T result)
+    {
+        result = T.Zero;
+
+        if (string.IsNullOrWhiteSpace(calculation))
+            return false;
+
+        var parts = calculation.Split(' ', StringSplitOptions.RemoveEmptyEntries);
+        if (parts.Length != 3)
+            return false;
+
+        var leftText = parts[0];
+        var opText = parts[1];
+        var rightText = parts[2];
+
+        if (opText.Length != 1)
+            return false;
+
+        var op = opText[0];
+
+        if (!T.TryParse(leftText, CultureInfo.InvariantCulture, out var left))
```

Comment:
nit: The Essential C# guideline says "AVOID omitting braces, except for the simplest of single-line if statements." All these single line if statements make me think what does simplest mean? I mean these are pretty simple but is the one on line 35 simpler?